### PR TITLE
refresh state after migrating update resources

### PR DESCRIPTION
**Pull request by ms-henglu**
Fixes https://github.com/Azure/azapi2azurerm/issues/60

### CODE REVIEW

File: tf/terraform.go [hunk 1]
```diff
@@ -238,6 +238,16 @@ func (t *Terraform) Apply() error {
 	return t.exec.Apply(context.TODO())
 }
 
+func (t *Terraform) RefreshState(resources []string) error {
+	opts := make([]tfexec.ApplyOption, 0)
+	// TODO: refresh-only option is not supported in terraform-exec, use tfexec.DirOrPlan as a hack
```
Comment (magodo) [inline tf/terraform.go:243]: Can we simply use `Refresh`?

Comment (ms-henglu) [reply tf/terraform.go:243]: Yes, but `refresh` command is deprecated: https://www.terraform.io/cli/commands/refresh#command-refresh

Comment (magodo) [reply tf/terraform.go:243]: Indeed, whilst probably still better than the hack used here?

**Review by magodo — APPROVED** [review]

File: tf/terraform.go (hunk 1 above)
Comment (ms-henglu) [reply tf/terraform.go:243]: Good point, I'll change it to refresh command.

**Pull request merged**